disnake/poll.py

```diff
@@ -0,0 +1,415 @@
+
+    You must specify at least one of the parameters when creating an instance.
+
+    .. versionadded:: 2.10
+
+    Parameters
+    ----------
+    text: Optional[:class:`str`]
+        The text of this media.
+    emoji: Optional[Union[:class:`Emoji`, :class:`PartialEmoji`, :class:`str`]]
+        The emoji of this media.
+
+    Attributes
+    ----------
+    text: Optional[:class:`str`]
+        The text of this media.
+    emoji: Optional[:class:`PartialEmoji`]
+        The emoji of this media.
+    """
+
+    __slots__ = ("text", "emoji")
+
+    def __init__(
+        self, text: Optional[str] = None, *, emoji: Optional[Union[Emoji, PartialEmoji, str]] = None
+    ) -> None:
+        if text is None and emoji is None:
+            raise ValueError("At least one of `text` or `emoji` must not be None")
+
+        self.text = text
+        self.emoji: Optional[Union[Emoji, PartialEmoji]] = None
+        if emoji is None:
+            self.emoji = None
+        elif isinstance(emoji, str):
+            self.emoji = PartialEmoji.from_str(emoji)
+        elif isinstance(emoji, _EmojiTag):
+            self.emoji = emoji
+        else:
+            raise TypeError("emoji must be None, a str, PartialEmoji, or Emoji instance.")
+
+    def __repr__(self) -> str:
+        return f"<{self.__class__.__name__} text={self.text!r} emoji={self.emoji!r}>"
+
+    @classmethod
+    def from_dict(cls, state: ConnectionState, data: PollMediaPayload) -> PollMedia:
+        text = data.get("text")
+
+        emoji = None
+        if emoji_data := data.get("emoji"):
+            emoji = state._get_emoji_from_data(emoji_data)
+
+        return cls(text=text, emoji=emoji)
+
+    def _to_dict(self) -> PollCreateMediaPayload:
+        payload: PollCreateMediaPayload = {}
+        if self.text:
+            payload["text"] = self.text
+        if self.emoji:
+            if isinstance(self.emoji, (Emoji, PartialEmoji)):
+                if self.emoji.id:
+                    payload["emoji"] = {"id": self.emoji.id}
+                else:
+                    payload["emoji"] = {"name": self.emoji.name}
+            else:
+                payload["emoji"] = {"name": self.emoji}
+        return payload
+
+
+class PollAnswer:
+    """Represents a poll answer from discord.
+
+    .. versionadded:: 2.10
+
+    Parameters
+    ----------
+    media: :class:`PollMedia`
+        The media object to set the text and/or emoji for this answer.
+
+    Attributes
+    ----------
+    id: Optional[:class:`int`]
+        The ID of the answer that this object represents. This will be None only when the object
+        is builded manually. The library always provide this attribute.
+    media: :class:`PollMedia`
+        The media fields linked to this answer.
+    poll: Optional[:class:`Poll`]
+        The poll that contain this answer. This will be None only when the object
+        is builded manually. The library always provide this attribute.
```

Comment:
```suggestion
    id: Optional[:class:`int`]
        The ID of this answer. This will be ``None`` only if this object was created manually
        and did not originate from the API.
    media: :class:`PollMedia`
        The media fields of this answer.
    poll: Optional[:class:`Poll`]
        The poll associated with this answer. This will be ``None`` only if this object was created manually
        and did not originate from the API.
```